react/features/recording/googleApi.js

```diff
@@ -50,142 +66,159 @@ const googleApi = {
-            globalGoogleApi
-            && globalGoogleApi.auth2
-            && globalGoogleApi.auth2.getAuthInstance
-            && globalGoogleApi.auth2.getAuthInstance().isSignedIn
-            && globalGoogleApi.auth2.getAuthInstance().isSignedIn.get()
+            googleApiClient
+            && googleApiClient.auth2
+            && googleApiClient.auth2.getAuthInstance
+            && googleApiClient.auth2.getAuthInstance().isSignedIn
+            && googleApiClient.auth2.getAuthInstance().isSignedIn.get()
         );
     },
 
     /**
-     * Loads the google api script.
+     * Generates a script tag and downloads the Google API Client Library.
      *
      * @returns {Promise}
      */
     load() {
-        if (gapiLoadPromise) {
-            return gapiLoadPromise;
+        if (googleClientLoadPromise) {
+            return googleClientLoadPromise;
         }
 
-        gapiLoadPromise = new Promise(resolve => {
+        googleClientLoadPromise = new Promise((resolve, reject) => {
             const scriptTag = document.createElement('script');
 
-            scriptTag.onload = resolve;
-            scriptTag.src = GOOGLE_JS_URL;
+            scriptTag.async = true;
+            scriptTag.addEventListener('error', () => {
+                scriptTag.removeEventListener('error');
+                scriptTag.removeEventListener('load');
+                scriptTag.remove();
+
+                googleClientLoadPromise = null;
 
-            document.body.appendChild(scriptTag);
+                reject();
+            });
+            scriptTag.addEventListener('load', resolve);
+            scriptTag.type = 'text/javascript';
+
+            scriptTag.src = GOOGLE_API_CLIENT_LIBRARY_URL;
+
+            document.head.appendChild(scriptTag);
         });
 
-        return gapiLoadPromise
+        return googleClientLoadPromise
             .then(() => new Promise((resolve, reject) =>
-                this.getGoogleApiGlobal().load('client:auth2', {
+                this._getGoogleApiClient().load('client:auth2', {
                     callback: resolve,
                     onerror: reject
                 })))
-            .then(() => this.getGoogleApiGlobal());
+            .then(() => this._getGoogleApiClient());
+    },
+
+    /**
+     * Prompts the participant to sign in to the Google API Client Library, even
+     * if already signed in.
+     *
+     * @returns {Promise}
+     */
+    forceSignIn() {
+        return this.get()
+            .then(api => api.auth2.getAuthInstance().signIn());
     },
 
     /**
-     * Makes a request for youtube broadcasts.
+     * Executes a request for a list of all YouTube broadcasts associated with
+     * user currently signed in to the Google API Client Library.
      *
      * @returns {Promise}
      */
-    requestAvailableBroadcasts() {
-        const url = _getLiveBroadcastsList();
+    requestAvailableYouTubeBroadcasts() {
+        const url = this._getURLForLiveBroadcasts();
```

Comment:
`URL` -> `Url`?

Reply:
At some point I was told to capitalize it as "URL" (and "ID"). I did a quick search just now and saw some functions within the react directory with a name that includes "Url" but the majority use "URL". So I'm going with URL.